runtime/termscounter_test.go

```diff
@@ -0,0 +1,151 @@
+package runtime_test
+
+import (
+	"go.etcd.io/gofail/runtime"
+	"testing"
+)
+
+// This variable mimics the code generated by gofail code package.
+// This works in tandem with exampleFunc function.
+var __fp_ExampleString *runtime.Failpoint //nolint:stylecheck
+
+// check if failpoint is initialized as gofail
+// tests can clear global variables of runtime packages
+func initFP() {
+	if __fp_ExampleString == nil { //nolint:stylecheck
+		__fp_ExampleString = runtime.NewFailpoint("ExampleString") //nolint:stylecheck
+	}
+}
```

Comment:
Why need these?

Reply:
> Two generic comments:
> 
> * As @lavacat mentioned, the design doc (and probably readme) need to be updated.
I will update this in the next checkin.

> * Please also consider to setup the e2e test ([Add e2e test #40](https://github.com/etcd-io/gofail/issues/40)), to verify the functionality end to end instead of just unit test; but of course, can be in a separate PR.
Sure, I will look at creating a e2e setup in a separate PR